transforms/transforms.go

```diff
@@ -215,3 +215,28 @@ func Unique(hash RowIdentifier) optimus.TransformFunc {
 		return false, nil
 	})
 }
+
+// GroupBy returns a TransformFunc that returns Rows of Rows grouped by id (see tests).
```

Comment:
Can we document the output format here, instead of referring people to the test cases?